### DISCUSSION TITLE
Cursor lags behind while any keyboard button is held when rendering with Metal

### DISCUSSION THREAD

**Discussion by thewildtree** [ppy/osu#23765] [Q&A]
### Type

Performance

### Bug description

_Disclaimer: I noticed this behaviour first around the renderer-selection update (2023.326.0?), but I switched to the legacy OpenGL renderer then and kept it that way until now. Reporting now because nothing has changed and the Metal renderer is supposed to be the lowest latency one now._

When using the **Metal** renderer, playing with a mouse, any keyboard keypress (especially if held, like on a slider) causes the cursor to suddenly slow down - like if you dramatically reduced the sensitivity - and then after a short while skip to where it's supposed to be if the lag didn't happen.
Only the cursor position is affected - the rest of the game doesn't even flinch and continues smoothly.

Seems to happen with all keys except for modifier ones (Control/Option etc.). I am currently unable to test whether tablet input is affected too.

Important note: with a 180Hz screen, this issue becomes very visible with all frame limiter options except VSync. With VSync, it definitely happens to a way lesser degree (can't immediately spot it when circling the cursor), but gameplay still feels weird compared to legacy OpenGL. Might be placebo, though. 
I know that with the latest update, options other than VSync won't do much cause the draw thread is locked anyway, but it would be nice to find the cause, because it might just be less visible there.

Playing on an MBP 16" with M1 Pro, on macOS Ventura 13.2.1.

The attached clips shows normal cursor movement first, and then laggy one when I start pressing/holding a key, with two skins to maybe make it more obvious. I hope they aren't too laggy to notice the issue, if they are I'll re-record with some better OBS settings. It's terribly more visible during gameplay, and makes the game pretty much unplayable with Metal. 

### Screenshots or videos

https://github.com/ppy/osu/assets/27722894/f5f7a7b5-1b01-4b7b-a935-e73181776c37

https://github.com/ppy/osu/assets/27722894/9a681c1b-0fd2-49d1-97ae-8553cba01285

### Version

2023.605.0

### Logs

[database.log](https://github.com/ppy/osu/files/11654178/database.log)
[input.log](https://github.com/ppy/osu/files/11654179/input.log)
[network.log](https://github.com/ppy/osu/files/11654180/network.log)
[performance.log](https://github.com/ppy/osu/files/11654181/performance.log)
[runtime.log](https://github.com/ppy/osu/files/11654182/runtime.log)

**Comment by peppy** [comment]
I can't see anything from the videos, and can't repro this locally.

Based on my knowledge of macOS, I think this is going to be something external to the game. Any app which handles input at a global level can add latency when things are going on.

Please go into system settings > accessibility and try removing all apps which have input capture abilities. If that fixes the issue, try enabling them again one-by-one until you find the app causing the issue.

**Reply by thewildtree** [reply]
Closing as this is clearly an early case of https://github.com/ppy/osu/issues/27227. 

Sorry about the late reply, but what you suggested didn't make a noticeable difference unfortunately.
